server/tree.js

```diff
@@ -0,0 +1,46 @@
+const path = require('path');
+const fs = require('fs');
+
+function isFile(content) {
+  return new Promise((resolve, reject) => {
+    fs.stat(content, (err, stats) => {
+      if (err) {
+        reject(err);
+      }
+
+      resolve(stats.isFile());
+    });
+  });
+}
+
+function makeAnswer(directoryContent, pathParam) {
+  return Promise.all(directoryContent.map(content => {
+    return isFile(path.resolve(pathParam, content))
+      .then(ifFile => {
+        return {
+          type: ifFile ? 'file' : 'directory',
+          name: content
+        };
+      });
+  }));
+}
+
+function getDirectoryContent(pathParam) {
+  return new Promise((resolve, reject) => {
+    if (!pathParam || !pathParam.trim().length) {
+      pathParam = path.resolve(__dirname, '../client/public');
+    }
+
+    fs.readdir(pathParam, (err, files) => {
```

Comment:
if you extracted `isFile` to function-promise wrapper maybe extract `readdir` as well? and then have this `getDirectoryContent` return call to function and not do `new Promise(…`?